spinalcordtoolbox/registration/register.py

```diff
@@ -134,6 +134,10 @@ def addStep(self, stepParam):
 def register_step_ants_slice_regularized_registration(src, dest, step, metricSize, fname_mask='', verbose=1):
     """
     """
+    # NB: If the mask is soft, fname_mask will be returned as None, so that it won't be further applied via '-x'
+    if fname_mask:
+        dest, fname_mask = image.apply_mask_if_soft(dest, fname_mask)
+
```

Comment:
This isn't the most elegant solution. The function feels opaque... it's hard to tell what's happening with the filenames here unless you go and look inside the function.

But, considering that this is a hacky workaround in the first place, I think it's more helpful to contain all of the logic inside a single function. I think it makes it a little easier to document? Also, having this logic contained in a one-liner like this makes it easier to potentially refactor it one day in the future.

Reply:
Don't sweat it, `moco` and `register` both need a major refactor.